server/leader.go

```diff
@@ -259,7 +302,51 @@ func (s *Server) watchLeader() {
 	}
 }
 
-func (s *Server) resignLeader() error {
+// ResignLeader resigns current PD's leadership. If nextLeader is empty, all
+// other pd-servers can campaign.
+func (s *Server) ResignLeader(nextLeader string) error {
+	// Determine next leaders.
+	var leaderIDs []string
+	res, err := etcdutil.ListEtcdMembers(s.client)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, member := range res.Members {
+		if (nextLeader == "" && member.ID != s.id) || (nextLeader != "" && member.Name == nextLeader) {
+			leaderIDs = append(leaderIDs, strconv.FormatUint(member.ID, 10))
+		}
+	}
+	nextLeaderValue := strings.Join(leaderIDs, ",")
+
+	// Save expect leader(s) to etcd.
+	lease := clientv3.NewLease(s.client)
+	defer lease.Close()
+	ctx, cancel := context.WithTimeout(s.client.Ctx(), requestTimeout)
+	leaseRes, err := lease.Grant(ctx, int64(nextLeaderTTL))
```

Comment:
We can make them the same. But this TTL doesn't have direct relationship with leader lease, it's ok to be different, no matter shorter or longer.